src/main/java/org/opentripplanner/middleware/otp/response/Itinerary.java

```diff
@@ -79,6 +79,39 @@
      */
     public List<Leg> legs = null;
 
+    /**
+     * Determines whether the itinerary includes transit.
+     * @return true if at least one {@link Leg} of the itinerary is a transit leg per OTP.
+     */
+    public boolean hasTransit() {
+        if (legs != null) {
+            for (Leg leg : legs) {
+                if (leg.transitLeg != null && leg.transitLeg) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Determines whether the itinerary includes transit.
+     * @return true if at least one {@link Leg} of the itinerary is a rental or ride hail leg per OTP.
+     */
+    public boolean hasRentalOrRideHail() {
+        if (legs != null) {
+            for (Leg leg : legs) {
+                if (leg.rentedBike != null && leg.rentedBike ||
+                    leg.rentedCar != null && leg.rentedCar ||
+                    leg.rentedVehicle != null && leg.rentedVehicle ||
+                    leg.hailedCar != null && leg.hailedCar) {
```

Comment:
Here and in the hasTransit method, I'm not crazy about all of the `null` checks.

It looks like this is an alternative approach `Boolean.TRUE.equals(leg.rentedBike)`. I'm somewhat in favor of this approach because it's a bit shorter and provides a more easily replicable approach for other cases we might have in the future.

Reply:
I like that better too, thanks for pointing out (see https://github.com/ibi-group/otp-middleware/pull/91/commits/134a1c76a195169fd9c0a2c196409b3b4664f57c). There are already some cases with null values in the tests (anywhere the `rented*/hailed` field is not set).